exporters/autoexport/exporter.go

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package autoexport // import "go.opentelemetry.io/contrib/exporters/autoexport"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/sdk/trace"
+)
+
+const otelTracesExportersEnvKey = "OTEL_EXPORTERS"
+
+// errUnknownExpoter is returned when an unknown exporter name is used in
+// the OTEL_EXPORTERS environment variable.
+var errUnknownExpoter = errors.New("unknown exporter")
+
+func NewTraceExporters(exporters ...trace.SpanExporter) []trace.SpanExporter {
```

Comment:
I think this needs to be defined for a single exporter.

```suggestion
func NewTraceExporter(exporter trace.SpanExporter) trace.SpanExporter {
```

That way it can be used similar to:

```go
tp := trace.NewTracerProvider(trace.WithBatcher(autoexport.NewTraceExporter(/* some defualt */)))
// ...
```

Reply:
Yeah, I agree. If we limit autoexport to only ever configure one exporter, it's a lot easier to reason about and removes the known limitation in the description (configuring multiple exporters with the same type).

Reply:
I've updated the func to only return a single SpanExporter, but the interface doesn't feel as nice. It was nice that the you could call autoexport.NewTraceExporter without a default exporter.